satpy/readers/file_handlers.py

```diff
@@ -157,7 +157,7 @@ def file_type_matches(self, ds_ftype):
         """
         if isinstance(ds_ftype, str) and ds_ftype == self.filetype_info['file_type']:
             return True
-        elif self.filetype_info['file_type'] in ds_ftype:
+        if isinstance(ds_ftype, (list, tuple)) and self.filetype_info['file_type'] in ds_ftype:
```

Comment:
Would it be clearer/simpler with this?
```python
        if isinstance(ds_ftype, str):
            ds_ftype = [ds_ftype]
        if self.filetype_info['file_type'] in ds_ftype:
```

Reply:
Or even
```python
        if isinstance(ds_ftype, str):
            ds_ftype = [ds_ftype]
        return self.filetype_info['file_type'] in ds_ftype
```

Reply:
Done. Along with the tests @pnuu asked for.